common/lib/xmodule/xmodule/tests/test_video.py

```diff
@@ -993,3 +1001,94 @@ def test_video_with_no_transcripts_translation_retrieval(self):
+
+        Args:
+            validation (StudioValidation): A validation object.
+            expected_msg (string): An expected validation message.
+        """
+        self.assertFalse(validation)  # Validation fails due to at least one warning/message
+        self.assertTrue(validation.summary)
+        self.assertEqual(StudioValidationMessage.WARNING, validation.summary.type)
+        self.assertIn(expected_msg, validation.summary.text)
+
+    @override_settings(ALL_LANGUAGES=ALL_LANGUAGES)
+    def test_no_transcript_validation_message(self):
+        """
+        Test the validation message when no associated transcript file uploaded.
+        """
+        # Test validation message for a single language that does not have transcript file.
+        xml_data_transcripts = '''
+            <video display_name="Test Video"
+                   youtube="1.0:p2Q6BrNhdh8,0.75:izygArpw-Qo,1.25:1EeWXzPdhSA,1.5:rABDYkeK0x8"
+                   show_captions="false"
+                   download_track="false"
+                   start_time="00:00:01"
+                   download_video="false"
+                   end_time="00:01:00">
+              <source src="http://www.example.com/source.mp4"/>
+              <track src="http://www.example.com/track"/>
+              <handout src="http://www.example.com/handout"/>
+              <transcript language="ur" src="" />
+            </video>
+        '''
+        expected_validation_msg = "There is no transcript file associated with {language_label} language.".format(
+            language_label='Urdu'
+        )
+        descriptor = instantiate_descriptor(data=xml_data_transcripts)
+        validation = descriptor.validate()
+        self.assert_validation_message(validation, expected_validation_msg)
+
+        # Test validation message for a multi languages that do not have transcript files.
+        xml_data_transcripts = '''
+            <video display_name="Test Video"
+                   youtube="1.0:p2Q6BrNhdh8,0.75:izygArpw-Qo,1.25:1EeWXzPdhSA,1.5:rABDYkeK0x8"
+                   show_captions="false"
+                   download_track="false"
+                   start_time="00:00:01"
+                   download_video="false"
+                   end_time="00:01:00">
+              <source src="http://www.example.com/source.mp4"/>
+              <track src="http://www.example.com/track"/>
+              <handout src="http://www.example.com/handout"/>
+              <transcript language="ur" src="" />
+              <transcript language="eo" src="" />
+            </video>
+        '''
+        expected_validation_msg = "There are no transcript files associated with {language_label} language.".format(
+            language_label='Urdu, Esperanto'
+        )
+        descriptor = instantiate_descriptor(data=xml_data_transcripts)
+        validation = descriptor.validate()
+        self.assert_validation_message(validation, expected_validation_msg)
```

Comment:
I would use ddt here for `<transcript />` to inject in  `xml_data_transcripts` instead of repeating same thing twice.